core/src/main/scala/org/apache/spark/deploy/master/ApplicationInfo.scala

```diff
@@ -43,6 +42,18 @@ private[spark] class ApplicationInfo(
   @transient var endTime: Long = _
   @transient var appSource: ApplicationSource = _
 
+  // A cap on the number of executors this application can have at any given time.
+  // By default, this is infinite. Only after the first allocation request is issued
+  // by the application will this be set to a finite value.
+  @transient var executorLimit: Int = _
```

Comment:
same reason, when this class is deserialized on the Master side, we set this variable to max value. It's not used on the driver side (before it's serialized) so there's no point initializing it here and duplicating the max value in `readObject`.